src/components/App.tsx

```diff
@@ -751,6 +752,35 @@ function App() {
     if (messages) navigator.clipboard.writeText(messages);
   };
 
+  /*//////////////////////////////////////////////////////////////
+                         RENAME NODE LOGIC
+  //////////////////////////////////////////////////////////////*/
+
+  // TODO: Prevent node name to be overridden when GPT node is edited
+
+  const [nodeToRename, setNodeToRename] = useState<Node>();
+
+  const toggleShowRenameInput = () => {
```

Comment:
this is kinda improperly named no? probably more aptly named `showRenameInput`, it doesn't toggle atm?